internal/cli/streams/processor/delete.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOpts struct {
+	cli.ProjectOpts
+	*cli.DeleteOpts
+	cli.StreamsOpts
+	store store.ProcessorDeleter
+}
+
+func (opts *DeleteOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *DeleteOpts) Run() error {
+	return opts.Delete(opts.store.DeleteStreamProcessor, opts.ProjectID, opts.Instance)
```

Comment:
```suggestion
	return opts.Delete(opts.store.DeleteStreamProcessor, opts.ConfigProjectID(), opts.Instance)
```
here and everywhere else you do `opts.ProjectID`, project id is layered, it can come from config, env or flag, using th emethod `ConfigProjectID()` will make sure to use the correct one other wise you are looking only to the flag values